script.js

```diff
@@ -1,49 +1,64 @@
-let msgWaiting = document.querySelector(".message-waiting");
-let searchForm = document.getElementById("search-form");
-
+const msgWaiting = document.querySelector(".message-waiting");
+const searchForm = document.getElementById("search-form");
 const booksDisplay = document.getElementById("books-display");
 
-let getGoogleBooks = "https://www.googleapis.com/books/v1/volumes?q=";
-
-let apiKey = "&key=AIzaSyCw6aeDwd2srAlaiVBFSSBvyl2O7atCTaA";
+const googleBooksURL = "https://www.googleapis.com/books/v1/volumes?q=";
+const apiKey = "&key=AIzaSyCw6aeDwd2srAlaiVBFSSBvyl2O7atCTaA";
 
 searchForm.addEventListener("submit", () => {
   msgWaiting.style.display = "none";
-  let searchInput = document.getElementById("searchInput").value;
-  let searchResult = `${getGoogleBooks}${searchInput}${apiKey}`;
-
-  async function getBooks() {
-    const booksPromise = await fetch(searchResult);
-    const getBook = await booksPromise.json();
-    booksDisplay.innerHTML = "";
-    getBook.items.map((book) => {
-      console.log(book.volumeInfo);
-
-      let title = book.volumeInfo.title;
-      let authors = book.volumeInfo.authors;
-      let previewLink = book.volumeInfo.previewLink;
-      let publishdate = book.volumeInfo.publishedDate;
-      let publisher = book.volumeInfo.publisher;
-      let smallThumbnail = book.volumeInfo.imageLinks.smallThumbnail;
-
-      let div = document.createElement("div");
-      div.innerHTML = `
-      
-      <h3>  ${title} </h3>
-      <div class= "book-img">
-       <img src= "${smallThumbnail}" alt= "Image book cover "> 
-       </div>
-        <div class = "book-details">
+  const searchInput = document.getElementById("search-input").value;
+  const searchQuery = `${googleBooksURL}${searchInput}${apiKey}`;
+
+  getBooks(searchQuery);
+});
+
+async function getBooks(query) {
+  const booksPromise = await fetch(query);
+  const booksData = await booksPromise.json();
+
+  displayBooks(booksData.items);
+}
+
+function displayBooks(books) {
+  // clear any previous results
+  booksDisplay.innerHTML = "";
+
+  books.forEach((book) => {
+
+    let volumeInfo = book.volumeInfo;
+    let title = volumeInfo.title || "N/A";
```

Comment:
I noticed that some search results don't contain data for all `volumeInfo` fields, and it was therefore showing up in the displayed results as "undefined," which would confuse non-web-developer users, and web developers would think it's a bug.

What this code does is ensure that if a `volumeInfo` field is undefined, that the text we show in the DOM is "N/A".

And how does this work? JavaScript executes Boolean expressions left to right. This means that `A || B` would be evaluated as follows:

- If `A` is truthy, then `A || B` will also be truthy regardless of what B is. JavaScript will therefore not even bother to evaluate `B`.
- If `A` if falsey, then `A || B` will evaluate to whatever `B` is, whether that's truthy or falsey. So the result of the expression `A || B` when `A` is falsey, is simply `B`.

We can therefore take advantage of the fact that `undefined` is falsey in JavaScript (as are a few other values – it's good to know what they all are).

So, if for example `volumeInfo.title` is `undefined`, then the value of `title` will be set to `"N/A"`.
If `volumeInfo.title` is `"Moon of the Crusted Snow"`, then the value of `title` will be set to `"Moon of the Crusted Snow"`.

Note: For testing purposes, try searching for 'Moon of the Crusted Snow' - some of the search results have incomplete information, hence how I noticed this particular bug.